operators/pkg/apis/common/v1alpha1/association.go

```diff
@@ -13,6 +13,7 @@ import (
 type AssociationStatus string
 
 const (
+	AssociationUnknown     AssociationStatus = ""
```

Comment:
Two questions:
1. Should this have an "Unknown" value?
2. Should we use this const instead of "" literal when returning (for example in `func (r *ReconcileApmServerElasticsearchAssociation) reconcileInternal(apmServer apmtype.ApmServer) (commonv1alpha1.AssociationStatus, error)` maybe elsewhere too)?

Reply:
Also, `func resultFromStatus(status commonv1alpha1.AssociationStatus) reconcile.Result` should probably be updated for completness.

Reply:
Good catch. I meant to do that but forgot.